arch/Kconfig

```diff
@@ -707,6 +707,11 @@ config CPU_HAS_DCACHE
 	help
 	  This hidden configuration should be selected when the CPU has a d-cache.
 
+config CPU_HAS_CACHE_COHERENCY
+	bool
+	help
+	  This hidden configuration should be selected when the CPU has cache coherency.
```

Comment:
Can we name this in the opposite direction: CPU_CACHE_INCOHERENT or something?  That's the notable state.  Almost everything is coherent.

Reply:
Also worth gating this, or at least pointing out in documentation, that we're specifically talking about intra-CPU multiprocessing incoherence.  The setting really only makes sense, even on intel_adsp, when MP_NUM_CPUS > 1.